python/cog/server/redis_queue.py

```diff
@@ -309,14 +278,15 @@ def handle_message(
 
                     if output_type.multi:
                         response["output"].append(output)
-                        send_response(response)
+                        yield response
                     else:
                         assert (
                             response["output"] is None
                         ), "Predictor unexpectedly returned multiple outputs"
                         response["output"] = output
 
                 elif isinstance(event, Done):
+                    # TODO assert we only get one Done
```

Comment:
I mean to be honest we can just do it here! `assert not done_event` should be fine here and we probably should crash the worker if it happens.